src/server/accept.rs

```diff
@@ -176,13 +189,83 @@ where
     }
 }
 
-async fn drain_connections(connections: &mut JoinSet<Result<(), DynError>>) {
+fn joined(result: &Result<TaskResult, JoinError>) {
+    match result {
+        // An error occurred that originates from hyper or tokio.
+        Ok(Err(error)) => {
+            if let Some(e) = error.downcast_ref::<hyper::Error>() {
+                let is_disconnect = e.is_canceled()
+                    || e.is_incomplete_message()
+                    || e.source().is_some_and(|source| {
+                        source
+                            .downcast_ref::<std::io::Error>()
+                            .is_some_and(|e| e.kind() == std::io::ErrorKind::NotConnected)
+                    });
+
+                if is_disconnect {
+                    // trace!();
+                } else {
+                    log!("error(http): {}", e);
+                }
+            } else {
+                log!("error(other): {}", error);
+            }
+        }
+        // The connection task panicked.
+        Err(error) if error.is_panic() => panic!("{}", error),
```

Comment:
If a task panics due to a poisoned mutex, we definitely want to panic on the main thread.

This seems less resilient but this is Rust, not Java. Restarting the main process from a supervisor is a best practice and one we should assume people are doing. This is what we would want regardless if we implemented the single threaded version of the server, first.